code/modules/mob/living/carbon/human/human.dm

```diff
@@ -69,6 +78,44 @@
 	popup.set_content(dat.Join("<br>"))
 	popup.open(FALSE)
 
+/mob/living/carbon/human/verb/show_gifts_self()
+	set category = "IC"
+	set name = "View Gifts"
+
+	ShowGifts()
+
+/mob/living/carbon/human/verb/show_gifts_other(mob/living/L in oview(1))
+	set category = "IC"
+	set name = "Show Gifts To"
+
+	if(istype(L))
+		if(do_after(src, 1 SECONDS, L))
+			ShowGifts(L)
+			emote("spin")
+			to_chat(src, "<span class='notice'>You have shown your Gifts to [L].</span>")
+			return
+		to_chat(src, "<span class='notice'>You must remain in place to show someone your Gifts!</span>")
+
+/mob/living/carbon/human/proc/ShowGifts(mob/viewer = src)
+	if(!LAZYLEN(ego_gift_list))
+		to_chat(viewer, "<span class='warning'>[src] has no Gifts!</span>")
+		return
+
+	var/list/dat = list()
+	dat += "<div align='center'><b>[real_name]</b></div><br>"
+	dat += "<div align='center'>Slot | Name ( Fortitude | Prudence | Temperance | Justice )</div><br>"
+	for(var/beta_gift in ego_gift_list) // These show the benefits of each one and at level 4+ you can lock the gift in place.
+		var/datum/ego_gifts/alpha_gift = ego_gift_list[beta_gift]
+		dat += "[alpha_gift.slot]: [alpha_gift.name] \
+			( [alpha_gift.fortitude_bonus >= 0 ? "+[alpha_gift.fortitude_bonus]":"[alpha_gift.fortitude_bonus]"] | \
+			[alpha_gift.prudence_bonus >= 0 ? "+[alpha_gift.prudence_bonus]":"[alpha_gift.prudence_bonus]"] | \
+			[alpha_gift.temperance_bonus >= 0 ? "+[alpha_gift.temperance_bonus]":"[alpha_gift.temperance_bonus]"] | \
+			[alpha_gift.justice_bonus >= 0 ? "+[alpha_gift.justice_bonus]":"[alpha_gift.justice_bonus]"] ) \
+			[get_user_level(src) > 3 ? "<A href='byond://?src=[REF(alpha_gift)]'>[alpha_gift.locked ? "Locked" : "Unlocked"]</A>" : ""]"
```

Comment:
Pressing that button won't refresh the window. Find a way to do so.

Reply:
Additionally, since you've added a proc to show it to the others, it will allow others to lock/unlock the gift slots.

Reply:
![image](https://user-images.githubusercontent.com/61567407/190234076-ac1202e1-7bb5-4750-88ec-729373371680.png)
![image](https://user-images.githubusercontent.com/61567407/190234124-4e311bb6-30f7-4470-948d-b10f1824f1f6.png)
First one is the "shown" version, second is the self-version of a level 5 agent

Reply:
Also added a variable that stores the owner a gift and uses that to call the "ShowGifts" proc so that it refreshes it when clicked. Works when tested.